packages/web/modals/validator-squad.tsx

```diff
@@ -98,22 +93,111 @@ export const ValidatorSquadModal: FunctionComponent<ValidatorSquadModalProps> =
       [totalStakePool]
     );
 
-    const rawData: Validator[] = useMemo(
+    const getFormattedVotingPower = useCallback(
+      (votingPower: Dec) =>
+        new RatePretty(votingPower)
+          .moveDecimalPointLeft(totalStakePool.currency.coinDecimals)
+          .maxDecimals(2)
+          .toString(),
+      [totalStakePool.currency.coinDecimals]
+    );
+
+    const getFormattedMyStake = useCallback(
+      (myStake) =>
+        new CoinPretty(totalStakePool.currency, myStake)
+          .maxDecimals(2)
+          .hideDenom(true)
+          .toString(),
+      [totalStakePool.currency]
+    );
+
+    const getCommissions = useCallback(
+      (validator: Staking.Validator) =>
+        new Dec(validator.commission.commission_rates.rate),
+      []
+    );
+
+    const getFormattedCommissions = useCallback(
+      (commissions: Dec) => new RatePretty(commissions)?.toString(),
+      []
+    );
+
+    const getIsAPRTooHigh = useCallback(
+      (commissions: Dec) => commissions.gt(new Dec(CONSTANTS.HIGH_APR)),
+      []
+    );
+
+    const getIsVotingPowerTooHigh = useCallback(
+      (votingPower: Dec) =>
+        new RatePretty(votingPower)
+          .moveDecimalPointLeft(totalStakePool.currency.coinDecimals)
+          .toDec()
+          .gt(new Dec(CONSTANTS.HIGH_VOTING_POWER)),
+      [totalStakePool.currency.coinDecimals]
+    );
+
+    const getFormattedWebsite = useCallback((website: string) => {
+      const displayUrl = normalizeUrl(website);
+      const truncatedDisplayUrl = truncateString(displayUrl, 30);
+      return truncatedDisplayUrl;
+    }, []);
+
+    const getImageUrl = useCallback(
+      (operator_address) =>
+        queryValidators.getValidatorThumbnail(operator_address),
+      [queryValidators]
+    );
+
+    const rawData: FormattedValidator[] = useMemo(
       () =>
         validators
           .filter(({ description }) => Boolean(description.moniker))
-          .map((validator) => ({
-            validatorName: validator.description.moniker,
-            myStake: getMyStake(validator),
-            votingPower: getVotingPower(validator),
-            commissions: new Dec(validator.commission.commission_rates.rate),
-            website: validator.description.website,
-            imageUrl: queryValidators.getValidatorThumbnail(
-              validator.operator_address
-            ),
-            operatorAddress: validator.operator_address,
-          })),
-      [validators, queryValidators, getVotingPower, getMyStake]
+          .map((validator) => {
+            const votingPower = getVotingPower(validator);
+            const myStake = getMyStake(validator);
+
+            const formattedVotingPower = getFormattedVotingPower(votingPower);
+            const formattedMyStake = getFormattedMyStake(myStake);
+
+            const commissions = getCommissions(validator);
```

Comment:
Are these functions necessary? You'll see this concept in the book I recommended, but IMO if the sole purpose of a function is to label what is happening while at the same time offering little functionality (here, just creating a pretty object, or calling another function), I'd rather have those operations be inline, then annotate it with a high level comment.

Then, I achieve the same effect without having a new developer jump back and forth, or reconcile how the function is being used throughout the codebase (i.e. how many times is it called, is it exported, what does it do, does it modify global vars, etc)

A general point inspired by this minor and inconsequential instance (this overhead is more pronounced across multiple files). Of course this will work fine, but when done often can be quite cumbersome to new devs.

Reply:
OK, this is fine in isolation, just a general philosophical point 👍